### PR TITLE
Profiler

### PR DESCRIPTION
**Pull request by Miguellissimo**
I closed the last PR because I messed up my rebase. To keep everything nice and clean I opened a new one.

### CODE REVIEW

**Comment by TheJJ** [comment]
Can you now add a keybinding or something for the profiler, so it isn't always run but just when some global mode is active?

**Comment by Miguellissimo** [comment]
I've made a few changes so that the profiler is only run in debug_overlay mode which can be toggled with the f3 key.

**Comment by TheJJ** [comment]
Nice!

>  if you have time, could you help me out with git rebase again. I think this command is not working quite well when a commit is already pushed to a public repository.

If you rewrote history, you need a force-push to "confirm" you want to discard the old commits.
so you do `git push origin profiler --force` ([see also](https://github.com/SFTtech/openage/blob/master/doc/contributing.md#rebasing---what-the-hell-is-it-and-why-do-i-need-it) at "After you have rebased stuff...")

File: copying.md [hunk 1]
```diff
@@ -57,6 +57,7 @@ _the openage authors_ are:
 | Franz-Niclas Muschter       | fm                          | fm@stusta.net                         |
 | Valentin Gagarin            | frickler01                  | valentin@fricklerhandwerk.de          |
 | Emmanouil Kampitakis        | madonius                    | emmanouil@kampitakis.de               |
+| Miguel Kasparick	      | miguellissimo		    | miguellissimo@gmail.com		    |
```
Comment (boombatower) [inline copying.md]: nit: off by a space.

Comment (TheJJ) [inline copying.md]: no, its indented with a tab, baaaaad @Miguellissimo, baaad.

**Comment by Miguellissimo** [comment]
fix of the rebasing and the small intentation issue.

**Comment by TheJJ** [comment]
øsøm!

**Pull request merged**